x-pack/solutions/security/plugins/security_solution/public/management/hooks/response_actions/use_get_pending_actions.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { UseQueryResult, UseQueryOptions } from '@tanstack/react-query';
+import { useQuery } from '@tanstack/react-query';
+import type { IHttpFetchError } from '@kbn/core-http-browser';
+import type { ResponseActionAgentType } from '../../../../common/endpoint/service/response_actions/constants';
+import { BASE_ENDPOINT_ACTION_ROUTE } from '../../../../common/endpoint/constants';
+import { useHttp } from '../../../common/lib/kibana';
+import type { ActionListApiResponse } from '../../../../common/endpoint/types';
+
+/**
+ * Request parameters for pending actions API
+ */
+export interface PendingActionsRequestQueryParams {
+  agentType?: ResponseActionAgentType;
+  endpointId?: string;
+  page?: number;
+  pageSize?: number;
+  commands?: string[];
+
+  [key: string]: string | number | string[] | undefined;
+}
+
+/**
+ * Error type for pending actions API errors
+ */
+export interface PendingActionsErrorType {
+  statusCode: number;
+  message: string;
+}
+
+/**
+ * Hook to retrieve pending response actions for cancellation
+ * Uses the standard actions list endpoint with pending status filter
+ * @param params - Query parameters including agentType, endpointId, etc.
+ * @param options - Additional options for the query
+ * @returns Query result containing pending actions data
+ */
+export const useGetPendingActions = (
```

Comment:
this i fine, but you could have just used teh existing hook - `useGetEndpointActionList()`